### PR TITLE
add: use fileutils.PatternMatcher for .dockerignore

### PR DESCRIPTION
**Pull request by nalind**
Use a fileutiles.PatternMatcher, which works better for this case than filepath.Match, to check if an item that we're copying is marked for exclusion by .dockerignore.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
@QiWang19 could you review this and give it a quick test run or three please?

**Comment by QiWang19** [comment]
@TomSweeneyRedHat LGTM

File: tests/bud/dockerignore3/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# comment
+*
+!test*
+!src
+**/*.in
+*.md
+!README*.md
+README-secret.md
+test1.txt
```
Comment (TomSweeneyRedHat) [inline tests/bud/dockerignore3/.dockerignore]: I'm not sure this is covered elsewhere, but could you create a `src/etc` directory with a couple of files in it and add 'src/etc to the end here and make sure those files don't get in?  Or do you think you've that scenario covered and I'm missing it?

**Comment by TomSweeneyRedHat** [comment]
@QiWang19 TYVM!

File: tests/bud/dockerignore3/.dockerignore (hunk 1 above)
Comment (nalind) [reply tests/bud/dockerignore3/.dockerignore]: It certainly can't hurt.  Added.

**Comment by TomSweeneyRedHat** [comment]
Thanks @nalind.  LGTM assuming happy tests.

**Comment by TomSweeneyRedHat** [comment]
Happy tests, going to merge this.

**Comment by TomSweeneyRedHat** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**